### PR TITLE
prevent numeric value navigation with data attr

### PR DESCRIPTION
**Pull request by johnbender**
This PR adds the ability to prevent numeric value navigation using the keyboard.

### CODE REVIEW

File: js/numeric-input.js [hunk 1]
```diff
@@ -25,9 +25,12 @@
 			}
 		}
 
+		this.isNavDisabled =
+			this.$el.attr("data-numeric-input-nav-disabled") !== null;
+
 		this.$el.on( "focus", function( e ) {
 			self.initMaxlength();
-		}).on( "keypress", function( e ) {
+		}).on( "keydown", function( e ) {
```
Comment (johnbender) [inline js/numeric-input.js]: This makes me nervous and needs at least one other set of eyes.

**Comment by toddparker** [comment]
@johnbender - I'll let @zachleat review the JS but I wonder if we could also key this off of the presence of this class which hides the spinner arrows:
`class="formcore-disable-spinner"`

...we can also have the data attr there, but if we added support via class this won't require a markup change

**Comment by zachleat** [comment]
LGTM @johnbender!

**Comment by toddparker** [comment]
@johnbender - see my comment above, might wan to add this class trigger as well.

**Pull request merged**